src/main/java/tc/oc/pgm/api/setting/SettingKey.java

```diff
@@ -78,4 +85,11 @@ public SettingValue getDefaultValue() {
   public String toString() {
     return getName();
   }
+
+  /**
+   * Called whether setting has changed and is ready to be updated internally.
+   *
+   * @param player owner of the setting
+   */
+  public void update(MatchPlayer player) {}
```

Comment:
For future reference, I think this will be a stop gap solution. In making `Datastore` async, I think I'll add support for a `Listener` to be registered.